aten/src/ATen/native/TypeProperties.cpp

```diff
@@ -75,8 +78,14 @@ ResultTypeState update_result_type_state(const Tensor& tensor, const ResultTypeS
   }
   ResultTypeState new_state = in_state;
   ScalarType current = tensor.scalar_type();
-  if (tensor.unsafeGetTensorImpl()->is_wrapped_number() && isFloatingType(current)) {
-    current = typeMetaToScalarType(at::get_default_dtype());
+  if (tensor.unsafeGetTensorImpl()->is_wrapped_number()) {
+    auto current_default = typeMetaToScalarType(at::get_default_dtype());
+    if(isComplexType(current)) {
+      current = (current_default == at::kFloat) ? at::kComplexFloat : at::kComplexDouble;
```

Comment:
we're probably going to want a centralized `get_default_complex_dtype()`, since this probably isn't the only place we have this kind of logic.